redfish-core/lib/log_services.hpp

```diff
@@ -3407,4 +3470,965 @@ inline void requestRoutesPostCodesEntry(App& app)
+                            << "from object: " << dbusObjIt->first.str;
+                        messages::internalError(asyncResp->res);
+                        break;
+                    }
+                    for (auto& assoc : *associations)
+                    {
+                        if (std::get<0>(assoc) == "isolated_hw")
+                        {
+                            fillIsolatedHwDetailsByObjPath(
+                                asyncResp,
+                                sdbusplus::message::object_path(
+                                    std::get<2>(assoc)),
+                                entryJsonIdx);
+                        }
+                        else if (std::get<0>(assoc) == "isolated_hw_errorlog")
+                        {
+                            sdbusplus::message::object_path errPath =
+                                std::get<2>(assoc);
+                            entryJson["AdditionalDataURI"] =
+                                "/redfish/v1/Systems/system/"
+                                "LogServices/EventLog/Entries/" +
+                                errPath.filename() + "/attachment";
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (entryIsPresent)
+    {
+        entryJson["@odata.type"] = "#LogEntry.v1_9_0.LogEntry";
+        entryJson["@odata.id"] =
+            "/redfish/v1/Systems/system/LogServices/HardwareIsolation/"
+            "Entries/" +
+            dbusObjIt->first.filename();
+        entryJson["Id"] = dbusObjIt->first.filename();
+        entryJson["Name"] = "Hardware Isolation Entry";
+        entryJson["EntryType"] = "Event";
+    }
+}
+
+/**
+ * @brief API Used to add the supported HardwareIsolation LogEntry Entries id
+ *
+ * @param[in] req - The HardwareIsolation redfish request (unused now).
+ * @param[in] asyncResp - The redfish response to return.
+ *
+ * @return The redfish response in the given buffer.
+ *
+ * @note This function will return the available entries dbus object which are
+ *       created by HardwareIsolation manager.
+ */
+inline void getSystemHardwareIsolationLogEntryCollection(
+    const crow::Request& /* req */,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+
+    auto getManagedObjectsHandler = [asyncResp](
+                                        const boost::system::error_code ec,
+                                        const GetManagedObjectsType& mgtObjs) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR
+                << "DBUS response error [" << ec.value() << " : "
+                << ec.message()
+                << "] when tried to get the HardwareIsolation managed objects";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        nlohmann::json& entriesArray = asyncResp->res.jsonValue["Members"];
+        if (!entriesArray.size())
+        {
+            entriesArray = nlohmann::json::array();
```

Comment:
Can you move this out of the if like 
https://github.com/openbmc/bmcweb/blob/9a128eb3fc3008f802974360e98de6495ed1a66b/redfish-core/include/utils/telemetry_utils.hpp#L41

Reply:
I don't think you need the if

Reply:
Done